File: Swift/LeetCode/LeetCode2492.swift
// https://leetcode.com/problems/minimum-score-of-a-path-between-two-cities

class Solution {
    func minScore(_ n: Int, _ roads: [[Int]]) -> Int {
        var graph = [Int: [(Int, Int)]]()
        var answer = Int.max
        var visited = Set<Int>()
        for road in roads {
            let (a, b, distance) = (road[0], road[1], road[2])
            graph[a, default: []].append((b, distance))
            graph[b, default: []].append((a, distance))
        }
        func dfs(current: Int) {
            visited.insert(current)
            for (nextNode, nextDistance) in graph[current] ?? [] {
                answer = answer > nextDistance ? nextDistance : answer
                if !visited.contains(nextNode) {
                    dfs(current: nextNode)
                }
            }
        }
        dfs(current: 1)
        return answer
    }
}

File: python/leetcode/leetcode2492.py
# https://leetcode.com/problems/minimum-score-of-a-path-between-two-cities

from collections import deque

class Solution:
    def minScore(self, n: int, roads: List[List[int]]) -> int:
        graph = defaultdict(list)
        for a, b, distance in roads:
            graph[a].append((distance, b))
            graph[b].append((distance, a))
        visited = set()
        answer = float("inf")
        stack = [1]
        while stack:
            node = stack.pop()
            if node in visited:
                continue
            visited.add(node)
            for distance, nextNode in graph[node]:
                ans = min(ans, distance)
                if nextNode not in visited:
                    stack.append(nextNode)

        return answer
